cogs/welcome/welcome.py

```diff
@@ -252,17 +252,145 @@ async def logServerLeave(self, leaveUser: discord.Member):
+        """Server welcome DM settings.
+
+        Upon joining the server, new members are sent a welcome DM.
+        Subcommands under this command control the content and behaviors of welcome DMs.
+        """
+
+    # [p]welcomeset dm message
+    @dm.command(name="message", aliases=["msg"])
+    async def setmessage(self, ctx: Context):
+        """Interactively configure the contents of the welcome DM."""
+        await ctx.send("What would you like the welcome DM message to be?")
+
+        def check(message: discord.Message):
+            return message.author == ctx.message.author and message.channel == ctx.message.channel
+
+        try:
+            message = await self.bot.wait_for("message", check=check, timeout=30.0)
+        except asyncio.TimeoutError:
+            await ctx.send("No response received, not setting anything!")
+            return
+
+        if len(message.content) > MAX_MESSAGE_LENGTH:
+            await ctx.send("Your message is too long!")
+            return
+
+        await self.config.guild(ctx.guild).get_attr(KEY_MESSAGE).set(message.content)
+        await ctx.send("Message set to:")
+        await ctx.send(f"```{message.content}```")
+        LOGGER.info(
+            "Message changed by %s#%s (%s)",
+            ctx.message.author.name,
+            ctx.message.author.discriminator,
+            ctx.message.author.id,
+        )
+        LOGGER.info(message.content)
+
+    # [p]welcomeset dm title
+    @dm.command(name="title")
+    async def setTitle(self, ctx: Context):
+        """Interactively configure the title for the welcome DM."""
+        await ctx.send("What would you like the welcome DM title to be?")
+
+        def check(message: discord.Message):
+            return message.author == ctx.message.author and message.channel == ctx.message.channel
+
+        try:
+            title = await self.bot.wait_for("message", check=check, timeout=30.0)
+        except asyncio.TimeoutError:
+            await ctx.send("No response received, not setting anything!")
+            return
+
+        if len(title.content) > 256:
+            await ctx.send("The title is too long!")
+            return
+
+        await self.config.guild(ctx.guild).get_attr(KEY_TITLE).set(title.content)
+        await ctx.send("Title set to:")
+        await ctx.send(f"```{title.content}```")
+        LOGGER.info(
+            "Title changed by %s#%s (%s)",
+            ctx.message.author.name,
+            ctx.message.author.discriminator,
+            ctx.message.author,
+        )
+        LOGGER.info(title.content)
+
+    # [p]welcomeset dm image
+    @dm.command(name="image")
+    async def setImage(self, ctx: Context, imageUrl: str = None):
+        """Sets an image in the embed with a URL.
+
+        Parameters:
+        -----------
+        imageUrl: str (optional)
+            The URL of the image to use in the DM embed. Leave blank to disable.
+        """
+        if imageUrl == "":
+            imageUrl = None
```

Comment:
Naive question: is this when someone explicitly does `[p]welcomeset dm image ""`? Can you double check and see if we can remove this?

Reply:
We can remove this block.